### ISSUE TITLE
Node 6.3.0 incompatibility

### ISSUE THREAD

**Issue by cbetta** [tschaub/mock-fs#137]
I just tried to use `mock-fs` in the just released Node 6.3.0 and getting this error:

```js
var mock = require('mock-fs');
mock();
fs.writeFileSync('test.txt', "test")
Error: ENOENT, no such file or directory 'test.txt'
    at Binding.<anonymous> (/project/node_modules/mock-fs/lib/binding.js:300:15)
    at maybeCallback (/project/node_modules/mock-fs/lib/binding.js:41:17)
    at Binding.open (/project/node_modules/mock-fs/lib/binding.js:274:10)
    at Object.fs.openSync (/project/node_modules/mock-fs/node/fs-6.0.0.js:584:18)
    at Object.fs.writeFileSync (/project/node_modules/mock-fs/node/fs-6.0.0.js:1224:33)
    at repl:1:4
    at sigintHandlersWrap (vm.js:32:31)
    at sigintHandlersWrap (vm.js:96:12)
    at ContextifyScript.Script.runInContext (vm.js:31:12)
    at REPLServer.defaultEval (repl.js:308:29)
```

**Comment by cbetta** [comment]
FWIW if I run the test suite I seem to get 143 errors.

**Comment by kotarou3** [comment]
This seems to be due to the following lines:
https://github.com/tschaub/mock-fs/blob/master/node/fs-6.0.0.js#L11
https://github.com/tschaub/mock-fs/blob/master/node/fs-6.0.0.js#L25

`process.binding('constants')` on nodejs v6.3.0 returns a hierarchical structure that is different to the flat structure that `require('constants')` would produce.
For example, `require('constants').O_CREAT === process.binding('constants').fs.O_CREAT`

I'm not sure why we're even using the non-public `process.binding()` call anyways...

**Comment by tschaub** [comment]
Thanks for the report.  Work in progress branch here: https://github.com/tschaub/mock-fs/compare/v6.3-fixes

**Comment by tschaub** [comment]
With the changes in #138, tests pass for me on 6.3.0, 5.11.0, and 4.2.4.  We'll see what Travis has to say about the other versions (a future release may drop all < 4).

**Comment by tschaub** [comment]
Fix published in `mock-fs@3.10.0`.
